spec/csv_row_model/import/represents_spec.rb

```diff
@@ -0,0 +1,207 @@
+require 'spec_helper'
+
+describe CsvRowModel::Import::Represents do
+  let(:klass) { Class.new(BasicImportModel) { def self.name; "RepresentsTestModel" end } }
+  let(:instance) { klass.new }
+
+  describe "instance" do
+    describe "#valid?" do
+      subject { instance.valid? }
+
+      it "calls #filter_errors" do
+        expect(instance).to receive(:filter_errors)
+        subject
+      end
+    end
+
+    describe "#filter_errors" do
+      subject { instance.send :filter_errors }
+      before do
+        klass.send(:represents_one, :test_model, dependencies: %i[string1]) { "test" }
+        instance.errors.add(:test_model)
+        instance.errors.add(:string2)
+      end
+
+      it "errors doesn't do anything with unrelated errors" do
+        subject
+        expect(instance.errors.keys).to eql [:test_model, :string2]
+      end
+
+      context "with an dependency errors" do
+        before { instance.errors.add(:string1) }
+
+        it "it removes the representation error" do
+          subject
+          expect(instance.errors.keys).to eql [:string2, :string1]
+        end
+
+        context "when called within #using_warnings" do
+          subject { instance.using_warnings { super() } }
+
+          before do
+            instance.warnings.add(:test_model)
+            instance.warnings.add(:string2)
+            instance.warnings.add(:string1)
+          end
+
+          it "removes warnings only" do
+            subject
+            expect(instance.errors.keys).to eql [:test_model, :string2, :string1]
+            expect(instance.warnings.keys).to eql [:string2, :string1]
```

Comment:
I'm not sure to understand why `instance.errors.keys` keep `:test_model ` representation error in this context....